### ISSUE TITLE
URL components are not urlencoded, resulting in API failures for objects with certain names

### ISSUE THREAD

**Issue by alienth** [sethvargo/go-fastly#14]
I have a cache condition with a name containing a character which needs to be urlencoded. (ex: a `text/html` condition). When I execute the following code which simply lists and tries to delete conditions which are found, it will fail:

```
        existingCacheSettings, err := client.ListCacheSettings(&fastly.ListCacheSettingsInput{Service: "abcd", Version: 10})
        for _, setting := range existingCacheSettings {
                err := client.DeleteCacheSetting(&fastly.DeleteCacheSettingInput{Service: "abcd", Name: setting.Name, Version: 10})
                if err != nil {
                        return err
                }
        }
```

The reason it is failing is because it ends up calling the following URL:

```
/service/abcd/version/10/condition/text/html
```

(note the unescaped forward-slash)

Any parameters which go into forming an API call URL should be urlescaped to prevent issues such as this. These object names are perfectly valid from Fastly's POV, but will cause go-fastly to malfunction.

I tried to address this in go-fastly by wrapping all URL components which could contain these characters in an escape method from `net/url`. Unfortunately, name components in the Fastly API may contain not only forward slashes but also whitespace, and `net/url` has no method to deal with both possible cases. Fixing this in `go-fastly` may require that we do a bit of a hacky `strings.Replace()`.

**Comment by alienth** [comment]
Bit of a stretch, but I've submitted a case to `net/url` to see if this type of escaping functionality can be supported: https://github.com/golang/go/issues/16207

**Comment by alienth** [comment]
For now I think the best option avail is to write a small function which calls `url.Parse()` on URL components, and also encodes the forward-slashes. Started work on it this evening. I'll plan on sending a PR tomorrow.

**Comment by sethvargo** [comment]
Looks like go 1.8 might fix this: https://github.com/golang/go/commit/7e2bf952a905f16a17099970392ea17545cdd193

**Comment by msuterski** [comment]
Did this ever got fixed (in go 1.8)?

I run into similar problem in Terraform 0.9.4 with conditions that included `/` in names.
